### ISSUE TITLE
`contains` allows empty array if `$ref` is used in sibling property.

### ISSUE THREAD

**Issue by naotokono** [ajv-validator/ajv#502]
**What version of Ajv are you using? Does the issue happen if you use the latest version?**
Latest (5.1.3)

**Ajv options object**

```javascript
options = {}

```

**JSON Schema**

```json
schema = {
   "definitions":{
      "def":{
         "type":"string"
      }
   },
   "type":"object",
   "properties":{
      "str":{
         "$ref":"#/definitions/def"
      },
      "arr":{
         "type":"array",
         "contains":{
            "type":"number"
         }
      }
   }
}

```

**Sample data**

```json
data = {"str":"", "arr":[]}

```

**Your code**

```javascript
ajv = new Ajv(options);
ajv.validate(schema, data)

```

**Validation result, data AFTER validation, error messages**

```
true
```

**What results did you expect?**
```
false
```

**Are you going to resolve the issue?**
No, I can't.

**Comment by epoberezkin** [comment]
@hogething interesting. And if you remove "str" from schema it correctly fails validation: https://runkit.com/esp/5926fa6b9ea69e0012c93625

**Comment by epoberezkin** [comment]
And it works in 4.x.x (with v5 option).

As a temp workaround, I can suggest replacing `contains: schema` with `not: {items: {not: schema}}`.

**Comment by naotokono** [comment]
What a quick response! Thanks, I will use the workaround for now and I hope the issue will be fixed soon :)

**Comment by epoberezkin** [comment]
And actually if "str" uses schema and not $ref it also works...: https://runkit.com/esp/5926fa6b9ea69e0012c93625

**Comment by epoberezkin** [comment]
option `{inlineRefs: false}` also solves the problem

**Comment by epoberezkin** [comment]
fixed in 5.1.4
